Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -710,11 +707,7 @@ static PlatformTimeRanges removeSamplesFromTrackBuffer(const DecodeOrderSampleMa
             trackBuffer.decodeQueue.erase(decodeKey);
 
         auto startTime = sample->presentationTime();
-#if USE(GSTREAMER)
```

Comment:
We've also been analyzing memory consumption recently and noticed the leak caused by the addition of the microsecond to endTime. The last sample was still stored in the TrackBuffer but not accounted in the PlatformTimeRanges, becoming virtually a leak.
Commit https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/8abb8ddba687991d32ce8fab2f2b752c14a8c92c in the https://github.com/WebPlatformForEmbedded/WPEWebKit/compare/eocanha/eocanha-debug-057-b work branch (not yet merged into wpe-20170728) gets rid of the microsecond addition. We had plans to clean and merge it at some point in the future.